### DISCUSSION TITLE
How to have DateTime as null?

### DISCUSSION THREAD

**Discussion by Azasratruth** [prisma/prisma#14886] [Q&A] [answered]
Model
```
model User {
    id           String    @id @default(uuid())
    createdAt    DateTime  @default(now())
    updatedAt    DateTime  @updatedAt
    deletedAt    DateTime?
    firstName    String
    lastName     String
    email        String    @unique
    passwordHash String
    phoneNumber  String?
    newsletter   Boolean?
}
```

Created a record via prisma studio, it adds `1970-01-01 00:00:00.000` in the `deletedAt` field rather than leaving it empty or putting null in it. Then I went to `phpmyadmin` and put null in place of the date. Came back to prisma studio and found this error.

```
Message: Error in Prisma Client request: 

Invalid `prisma.user.findMany()` invocation:

  Error converting field "deletedAt" of expected non-nullable type "DateTime", found incompatible value of "null".
  
Query:
{
  "modelName": "User",
  "operation": "findMany",
  "args": {
    "take": 100,
    "skip": 0,
    "select": {
      "id": true,
      "createdAt": true,
      "updatedAt": true,
      "deletedAt": true,
      "firstName": true,
      "lastName": true,
      "email": true,
      "passwordHash": true,
      "phoneNumber": true,
      "newsletter": true,
    }
  }
}
```  

I could work around this but I'd prefer to have it nullable.

**Comment by ludralph** [comment]
Hi @Haise17  👋 

I tried reproducing your exact scenario but couldn't. I created a record using Prisma studio on the same schema you provided and I can confirm that Prisma studio uses `null` for the deletedAt field if no value was inserted in that field.
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/13855608/185606204-72d7844b-6bd9-4c61-9736-b2e579dcf39a.png">

Can you also share the record you inserted?

**Reply by Azasratruth** [reply]
I had removed the column and moved on. I recreated it to get the record, and now interestingly, it's working properly. Appreciate you looking into it though.

**Comment by Azasratruth** [comment, marked as answer]
Closing this as it has resolved itself.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
